### PR TITLE
Preliminary PR to remove the fixed three-byte header pending

### PR DESCRIPTION
**Pull request by ekr**
measurements.

This is a placeholder for discussion.

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -3264,17 +3273,18 @@ by an encrypted body, which itself contains a type and optional padding.
...
 
-type
-: The content type of the record.
+:masked_length
+
+The length value with the high bit set. Records are always less than
+2^15 bytes, so this bit is unused. Having the high bit set allows
+receivers to easily distinguish TLS 1.3 encrypted packets from
+TLS 1.2 compatible framed plaintext packets (which start with the
+content type).
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: Might want to point out that plaintext packets are always either alerts or handshake content types, which don't set the high bit.

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -3264,17 +3273,18 @@ by an encrypted body, which itself contains a type and optional padding.
        } TLSInnerPlaintext;
 
        struct {
-           ContentType opaque_type = 23; /* application_data */
-           ProtocolVersion legacy_record_version = 0x0301; /* TLS v1.x */
-           uint16 length;
+           uint16 masked_length;
```
Comment (tmshort) [inline draft-ietf-tls-tls13.md]: A 32-bit header would be better for crypto off-loading, but 16-bits is better than 5 bytes.

**Comment by mimoo** [comment]
Why not using a `ContentType encrypted_type = 24` instead of `masked_length`.

This would make it easier to differentiate it from a `TLSPlaintext` message as well.

**Comment by ekr** [comment]
Well, this PR is intended to remove that byte entirely.

You still need to have legacy_record_version in unencrypted records.

**Comment by mimoo** [comment]
> Well, this PR is intended to remove that byte entirely.

I just find it cleaner (and it would add some sense to the specification as well) compared to a "hack" bit inside of an already used `uint16`. But I have no real argument for it :)

> You still need to have legacy_record_version in unencrypted records.

Right, although there are still some mentions to `TLSCiphertext.legacy_record_version`.

**Comment by ekr** [comment]
Closed per WG consensus in Chicago

**Pull request closed without merging**